wallet/wallet.c

```diff
@@ -882,6 +882,38 @@ static unsigned int count_trailing_zeroes(uint64_t index)
 #endif
 }
 
+static void wallet_stub_shachain_init(struct wallet *wallet,
+				      struct wallet_shachain *chain)
+{
+	struct db_stmt *stmt;
+	stmt = db_prepare_v2(
+		wallet->db,
+		SQL("INSERT into shachains (min_index, num_valid) VALUES (?, ?);"));
+	db_bind_u64(stmt, chain->chain.min_index);
+	db_bind_u64(stmt, chain->chain.num_valid);
+	db_exec_prepared_v2(stmt);
+
+	chain->id = db_last_insert_id_v2(stmt);
+	tal_free(stmt);
+
+	for (u32 pos = 0; pos < chain->chain.num_valid; pos++) {
+		struct secret s;
+		memcpy(&s, &chain->chain.known[pos].hash, sizeof(s));
```

Comment:
We have a new macro for, this: CROSS_TYPE_ASSIGNMENT(&s, &chain->chain.known[pos].hash).  It checks the two types are same size...